src/tests/src/solver/utils/test_ortools_quadratic_wrapper.cpp

```diff
@@ -0,0 +1,341 @@
+                   const std::vector<double>& coefs)
+    {
+        ++problemeAResoudre.NombreDeContraintes;
+        problemeAResoudre.NomDesContraintes.emplace_back(name);
+        std::string sens;
+        if (lb == ub)
+        {
+            problemeAResoudre.Sens += "=";
+            problemeAResoudre.SecondMembre.emplace_back(lb);
+        }
+        else if (std::isinf(lb))
+        {
+            problemeAResoudre.Sens += "<";
+            problemeAResoudre.SecondMembre.emplace_back(ub);
+        }
+        else if (std::isinf(ub))
+        {
+            problemeAResoudre.Sens += ">";
+            problemeAResoudre.SecondMembre.emplace_back(lb);
+        }
+        else
+        {
+            throw std::invalid_argument(
+              "Constraint lb and ub must be equal, or one of them infinite");
+        }
+        int iStart = problemeAResoudre.CoefficientsDeLaMatriceDesContraintes.size();
+        problemeAResoudre.IndicesDebutDeLigne.emplace_back(iStart);
+        problemeAResoudre.NombreDeTermesDesLignes.emplace_back(coefs.size());
+        for (int i = 0; i < coefs.size(); ++i)
+        {
+            problemeAResoudre.IndicesColonnes[iStart + i] = vars[i];
+            problemeAResoudre.CoefficientsDeLaMatriceDesContraintes[iStart + i] = coefs[i];
+        }
+    }
+
+    void checkPrimalSolution(const std::vector<double>& expected)
+    {
+        BOOST_TEST(problemeAResoudre.X == expected,
+                   boost::test_tools::tolerance(tolerance)
+                     << "comparison failed" << boost::test_tools::per_element());
+        BOOST_TEST(primals == expected,
+                   boost::test_tools::tolerance(tolerance)
+                     << "comparison failed" << boost::test_tools::per_element());
+    }
+
+    void checkDualSolution(const std::vector<double>& expected)
+    {
+        BOOST_TEST(problemeAResoudre.CoutsMarginauxDesContraintes == expected,
+                   boost::test_tools::tolerance(tolerance)
+                     << "comparison failed" << boost::test_tools::per_element());
+        BOOST_TEST(duals == expected,
+                   boost::test_tools::tolerance(tolerance)
+                     << "comparison failed" << boost::test_tools::per_element());
+    }
+
+    void checkReducedCosts(const std::vector<double>& expected)
+    {
+        BOOST_TEST(problemeAResoudre.CoutsReduits == expected,
+                   boost::test_tools::tolerance(tolerance)
+                     << "comparison failed" << boost::test_tools::per_element());
+        BOOST_TEST(reducedCosts == expected,
+                   boost::test_tools::tolerance(tolerance)
+                     << "comparison failed" << boost::test_tools::per_element());
+    }
+
+    void checkAllNan(const std::vector<double>& actual)
+    {
+        for (int i = 0; i < actual.size(); ++i)
+        {
+            BOOST_TEST(std::isnan(actual[i]));
+        }
+    }
+};
+
+BOOST_AUTO_TEST_SUITE(tests_on_ortools_quadratic_wrapper)
```

Comment:
As all your unit tests use the same fixture, you could use : 
```c++
BOOST_FIXTURE_TEST_SUITE(tests_on_ortools_quadratic_wrapper, QpFixture);

BOOST_AUTO_TEST_CASE(test_case1)
{...}

BOOST_AUTO_TEST_CASE(test_case2)
{...}
...
BOOST_AUTO_TEST_SUITE_END()
```
As you can see, no need to pass the fixture for each test case